docs/source-pytorch/cli/lightning_cli_advanced_3.rst

```diff
@@ -209,69 +213,15 @@ If the CLI is implemented as :code:`LightningCLI(MyMainModel)` the configuration
-:code:`class_path`.
-
-
-Class type defaults
-^^^^^^^^^^^^^^^^^^^
-
-The support for classes as type hints allows to try many possibilities with the same CLI. This is a useful feature, but
-it can make it tempting to use an instance of a class as a default. For example:
-
-.. code-block::
-
-    class MyMainModel(LightningModule):
-        def __init__(
-            self,
-            backbone: torch.nn.Module = MyModel(encoder_layers=24),  # BAD PRACTICE!
-        ):
-            super().__init__()
-            self.backbone = backbone
-
-Normally classes are mutable as it is in this case. The instance of :code:`MyModel` would be created the moment that the
-module that defines :code:`MyMainModel` is first imported. This means that the default of :code:`backbone` will be
-initialized before the CLI class runs :code:`seed_everything` making it non-reproducible. Furthermore, if
-:code:`MyMainModel` is used more than once in the same Python process and the :code:`backbone` parameter is not
-overridden, the same instance would be used in multiple places which very likely is not what the developer intended.
-Having an instance as default also makes it impossible to generate the complete config file since for arbitrary classes
-it is not known which arguments were used to instantiate it.
-
-A good solution to these problems is to not have a default or set the default to a special value (e.g. a
-string) which would be checked in the init and instantiated accordingly. If a class parameter has no default and the CLI
-is subclassed then a default can be set as follows:
-
-.. testcode::
-
-    default_backbone = {
-        "class_path": "import.path.of.MyModel",
-        "init_args": {
-            "encoder_layers": 24,
-        },
-    }
-
-
-    class MyLightningCLI(LightningCLI):
-        def add_arguments_to_parser(self, parser):
-            parser.set_defaults({"model.backbone": default_backbone})
+It is also possible to combine ``subclass_mode_model=True`` and submodules, thereby having two levels of ``class_path``.
 
-A more compact version that avoids writing a dictionary would be:
-
-.. testcode::
-
-    from jsonargparse import lazy_instance
-
-
-    class MyLightningCLI(LightningCLI):
-        def add_arguments_to_parser(self, parser):
-            parser.set_defaults({"model.backbone": lazy_instance(MyModel, encoder_layers=24)})
 
 Optimizers
 ^^^^^^^^^^
 
-If you will not be changing the class, you can manually add the arguments for specific optimizers and/or
-learning rate schedulers by subclassing the CLI. This has the advantage of providing the proper help message for those
-classes. The following code snippet shows how to implement it:
+In some case it might be desired to fix the optimizer and/or learning scheduler scheduler instead of allowing multiple.
+For this you can manually add the arguments for specific classes by subclassing the CLI. The following code snippet
+shows how to implement it:
```

Comment:
```suggestion
In some cases, fixing the optimizer and/or learning scheduler might be desired instead of allowing multiple.
For this, you can manually add the arguments for specific classes by subclassing the CLI. The following code snippet
shows how to implement it:
```